### PR TITLE
change OutBuffer* to ref OutBuffer

### PR DESCRIPTION
**Pull request by WalterBright**
Tedious, but simple.

### CODE REVIEW

File: compiler/src/dmd/frontend.h [hunk 1]
```diff
@@ -2891,7 +2891,7 @@ enum : int32_t { TEST_EMIT_ALL = 0 };
 
 extern void genhdrfile(Module* m, OutBuffer& buf);
 
-extern void moduleToBuffer(OutBuffer* buf, Module* m);
+extern void moduleToBuffer(OutBuffer& buf, Module* m);
```
Comment (ibuclaw) [inline compiler/src/dmd/frontend.h:2894]: I'd expect there to be another copy of this function in the hand-maintained headers used by gdc/ldc.

**Review by ibuclaw — DISMISSED** [review]

File: compiler/src/dmd/frontend.h (hunk 1 above)
Comment (WalterBright) [reply compiler/src/dmd/frontend.h:2894]: There is. Fixed.

**Comment by RazvanN7** [comment]
```
src/dmd/frontend.d(459): Error: function `dmd.hdrgen.moduleToBuffer2(Module m, ref OutBuffer buf, HdrGenState* hgs)` is not callable using argument types `(Module, OutBuffer*, HdrGenState*)`
src/dmd/frontend.d(459):        cannot pass rvalue argument `& buf` of type `OutBuffer*` to parameter `ref OutBuffer buf`
```

**Comment by WalterBright** [comment]
fixed

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**